### ISSUE TITLE
Tried with flatpak this time, new error

### ISSUE THREAD

**Issue by samuk190** [TheLastRar/CLR-DEV9#34]
Init Mono Runtime
Loaded Plugin
The type initializer for 'Sys' threw an exception.
  at System.IO.FileSystem.FileExists (System.ReadOnlySpan`1[T] fullPath, System.Int32 fileType, Interop+ErrorInfo& errorInfo) [0x00007] in <9f0df102fe6e4cfea29d2e46f585d8a5>:0 
  at System.IO.FileSystem.FileExists (System.ReadOnlySpan`1[T] fullPath) [0x00006] in <9f0df102fe6e4cfea29d2e46f585d8a5>:0 
  at System.IO.File.Exists (System.String path) [0x00043] in <9f0df102fe6e4cfea29d2e46f585d8a5>:0 
  at PSE.CLR_PSE_PluginLog.Open (System.String logFolderPath, System.String logFileName, System.String prefix, System.Collections.Generic.Dictionary`2[TKey,TValue] sourceIDs) [0x00078] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at CLRDEV9.CLR_DEV9.LogSetup () [0x00051] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at CLRDEV9.CLR_DEV9.Init () [0x00000] in <fd03a64f865b42448fec49d2c9d711c5>:0 
InnitError

Unhandled Exception:
System.TypeInitializationException: The type initializer for 'System.Windows.Forms.WindowsFormsSynchronizationContext' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.Windows.Forms.ThemeEngine' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.Drawing.GDIPlus' threw an exception. ---> System.DllNotFoundException: libc assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) System.Drawing.GDIPlus.uname(intptr)
  at System.Drawing.GDIPlus..cctor () [0x0006e] in <48f39c7a28e14dde88053e621a43853e>:0 
   --- End of inner exception stack trace ---
  at System.Drawing.StringFormat..ctor (System.Drawing.StringFormatFlags options, System.Int32 language) [0x00011] in <48f39c7a28e14dde88053e621a43853e>:0 
  at System.Drawing.StringFormat..ctor () [0x00000] in <48f39c7a28e14dde88053e621a43853e>:0 
  at (wrapper remoting-invoke-with-check) System.Drawing.StringFormat..ctor()
  at System.Windows.Forms.ThemeWin32Classic.ResetDefaults () [0x0001f] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.ThemeWin32Classic..ctor () [0x00006] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.ThemeEngine..cctor () [0x00012] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
   --- End of inner exception stack trace ---
  at System.Windows.Forms.SystemInformation.get_MenuAccessKeysUnderlined () [0x00000] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.Control..ctor () [0x000d6] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control..ctor()
  at System.Windows.Forms.WindowsFormsSynchronizationContext..cctor () [0x00000] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
   --- End of inner exception stack trace ---
  at System.Windows.Forms.Control..ctor () [0x0000d] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.ScrollableControl..ctor () [0x00000] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.ContainerControl..ctor () [0x0000e] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.Form..ctor () [0x00012] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.MessageBox+MessageBoxForm..ctor (System.Windows.Forms.IWin32Window owner, System.String text, System.String caption, System.Windows.Forms.MessageBoxButtons buttons, System.Windows.Forms.MessageBoxIcon icon, System.Boolean displayHelpButton) [0x0000c] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.MessageBox+MessageBoxForm..ctor (System.Windows.Forms.IWin32Window owner, System.String text, System.String caption, System.Windows.Forms.MessageBoxButtons buttons, System.Windows.Forms.MessageBoxIcon icon) [0x00000] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.MessageBox+MessageBoxForm..ctor(System.Windows.Forms.IWin32Window,string,string,System.Windows.Forms.MessageBoxButtons,System.Windows.Forms.MessageBoxIcon)
  at System.Windows.Forms.MessageBox.Show (System.String text) [0x00000] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at PSE.CLR_PSE_PluginLog.MsgBoxErrorTrapper (System.Exception e) [0x0003b] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at CLRDEV9.CLR_DEV9.Log_Fatal (System.Exception ex) [0x00000] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at CLRDEV9.CLR_DEV9.Init () [0x00054] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at PSE.CLR_PSE_DEV9.DEV9init () [0x00000] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at (wrapper native-to-managed) PSE.CLR_PSE_DEV9.DEV9init(System.Exception&)

When trying to load the plugin in pcsx2 flatpak (linux 1.6.0), what could be?

**Comment by TheLastRar** [comment]
Can you run `ldd /usr/lib32/pcsx2/mono_i386/usr/lib/libgdiplus.so`
Exact path will depend on where flatpak installed PCSX2

**Comment by samuk190** [comment]
```
ldd: aviso: você não tem permissão para execução para `/var/lib/flatpak/app/net.pcsx2.PCSX2/current/active/files/lib32/pcsx2/mono_i386/usr/lib/libgdiplus.so'
/var/lib/flatpak/app/net.pcsx2.PCSX2/current/active/files/lib32/pcsx2/mono_i386/usr/lib/libgdiplus.so: /usr/lib32/libtiff.so.5: no version information available (required by /var/lib/flatpak/app/net.pcsx2.PCSX2/current/active/files/lib32/pcsx2/mono_i386/usr/lib/libgdiplus.so)
	linux-gate.so.1 (0xf7f78000)
	libglib-2.0.so.0 => /usr/lib32/libglib-2.0.so.0 (0xf7d66000)
	libcairo.so.2 => /usr/lib32/libcairo.so.2 (0xf7c13000)
	libfreetype.so.6 => /usr/lib32/libfreetype.so.6 (0xf7b3b000)
	libjpeg.so.8 => /usr/lib32/libjpeg.so.8 (0xf7a93000)
	libtiff.so.5 => /usr/lib32/libtiff.so.5 (0xf7a0d000)
	libgif.so.7 => /usr/lib32/libgif.so.7 (0xf7a01000)
	libpng16.so.16 => /usr/lib32/libpng16.so.16 (0xf79c4000)
	libX11.so.6 => /usr/lib32/libX11.so.6 (0xf7875000)
	libexif.so.12 => not found
	libpthread.so.0 => /usr/lib32/libpthread.so.0 (0xf7853000)
	libm.so.6 => /usr/lib32/libm.so.6 (0xf7786000)
	libfontconfig.so.1 => /usr/lib32/libfontconfig.so.1 (0xf7734000)
	libc.so.6 => /usr/lib32/libc.so.6 (0xf7542000)
	libpcre.so.1 => /usr/lib32/libpcre.so.1 (0xf74cb000)
	libpixman-1.so.0 => /usr/lib32/libpixman-1.so.0 (0xf741a000)
	libxcb-shm.so.0 => /usr/lib32/libxcb-shm.so.0 (0xf7415000)
	libxcb.so.1 => /usr/lib32/libxcb.so.1 (0xf73e9000)
	libxcb-render.so.0 => /usr/lib32/libxcb-render.so.0 (0xf73d9000)
	libXrender.so.1 => /usr/lib32/libXrender.so.1 (0xf73cb000)
	libXext.so.6 => /usr/lib32/libXext.so.6 (0xf73b5000)
	libz.so.1 => /usr/lib32/libz.so.1 (0xf739b000)
	libbz2.so.1.0 => /usr/lib32/libbz2.so.1.0 (0xf7389000)
	libharfbuzz.so.0 => /usr/lib32/libharfbuzz.so.0 (0xf72af000)
	libzstd.so.1 => /usr/lib32/libzstd.so.1 (0xf71da000)
	liblzma.so.5 => /usr/lib32/liblzma.so.5 (0xf71ae000)
	libdl.so.2 => /usr/lib32/libdl.so.2 (0xf71a8000)
	/usr/lib/ld-linux.so.2 (0xf7f7a000)
	libexpat.so.1 => /usr/lib32/libexpat.so.1 (0xf717b000)
	libXau.so.6 => /usr/lib32/libXau.so.6 (0xf7176000)
	libXdmcp.so.6 => /usr/lib32/libXdmcp.so.6 (0xf716e000)
```
I saw "No permission ..." then I did sudo chmod 777 -R mono and sudo chmod 777 libCLR .so

after that:
`/var/lib/flatpak/app/net.pcsx2.PCSX2/current/active/files/lib32/pcsx2/mono_i386/usr/lib/libgdiplus.so: /usr/lib32/libtiff.so.5: no version information available (required by /var/lib/flatpak/app/net.pcsx2.PCSX2/current/active/files/lib32/pcsx2/mono_i386/usr/lib/libgdiplus.so)
	linux-gate.so.1 (0xf7f3d000)
	libglib-2.0.so.0 => /usr/lib32/libglib-2.0.so.0 (0xf7d2b000)
	libcairo.so.2 => /usr/lib32/libcairo.so.2 (0xf7bd8000)
	libfreetype.so.6 => /usr/lib32/libfreetype.so.6 (0xf7b00000)
	libjpeg.so.8 => /usr/lib32/libjpeg.so.8 (0xf7a58000)
	libtiff.so.5 => /usr/lib32/libtiff.so.5 (0xf79d2000)
	libgif.so.7 => /usr/lib32/libgif.so.7 (0xf79c6000)
	libpng16.so.16 => /usr/lib32/libpng16.so.16 (0xf7989000)
	libX11.so.6 => /usr/lib32/libX11.so.6 (0xf783a000)
	libexif.so.12 => not found
	libpthread.so.0 => /usr/lib32/libpthread.so.0 (0xf7818000)
	libm.so.6 => /usr/lib32/libm.so.6 (0xf774b000)
	libfontconfig.so.1 => /usr/lib32/libfontconfig.so.1 (0xf76f9000)
	libc.so.6 => /usr/lib32/libc.so.6 (0xf7507000)
	libpcre.so.1 => /usr/lib32/libpcre.so.1 (0xf7490000)
	libpixman-1.so.0 => /usr/lib32/libpixman-1.so.0 (0xf73df000)
	libxcb-shm.so.0 => /usr/lib32/libxcb-shm.so.0 (0xf73da000)
	libxcb.so.1 => /usr/lib32/libxcb.so.1 (0xf73ae000)
	libxcb-render.so.0 => /usr/lib32/libxcb-render.so.0 (0xf739e000)
	libXrender.so.1 => /usr/lib32/libXrender.so.1 (0xf7390000)
	libXext.so.6 => /usr/lib32/libXext.so.6 (0xf737a000)
	libz.so.1 => /usr/lib32/libz.so.1 (0xf7360000)
	libbz2.so.1.0 => /usr/lib32/libbz2.so.1.0 (0xf734e000)
	libharfbuzz.so.0 => /usr/lib32/libharfbuzz.so.0 (0xf7274000)
	libzstd.so.1 => /usr/lib32/libzstd.so.1 (0xf719f000)
	liblzma.so.5 => /usr/lib32/liblzma.so.5 (0xf7173000)
	libdl.so.2 => /usr/lib32/libdl.so.2 (0xf716d000)
	/usr/lib/ld-linux.so.2 (0xf7f3f000)
	libexpat.so.1 => /usr/lib32/libexpat.so.1 (0xf7140000)
	libXau.so.6 => /usr/lib32/libXau.so.6 (0xf713b000)
	libXdmcp.so.6 => /usr/lib32/libXdmcp.so.6 (0xf7133000)
`
New error(?):
```
`Init Mono Runtime
Loaded Plugin
The type initializer for 'Sys' threw an exception.
  at System.IO.FileSystem.FileExists (System.ReadOnlySpan`1[T] fullPath, System.Int32 fileType, Interop+ErrorInfo& errorInfo) [0x00007] in <9f0df102fe6e4cfea29d2e46f585d8a5>:0 
  at System.IO.FileSystem.FileExists (System.ReadOnlySpan`1[T] fullPath) [0x00006] in <9f0df102fe6e4cfea29d2e46f585d8a5>:0 
  at System.IO.File.Exists (System.String path) [0x00043] in <9f0df102fe6e4cfea29d2e46f585d8a5>:0 
  at PSE.CLR_PSE_PluginLog.Open (System.String logFolderPath, System.String logFileName, System.String prefix, System.Collections.Generic.Dictionary`2[TKey,TValue] sourceIDs) [0x00078] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at CLRDEV9.CLR_DEV9.LogSetup () [0x00051] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at CLRDEV9.CLR_DEV9.Init () [0x00000] in <fd03a64f865b42448fec49d2c9d711c5>:0 
InnitError

Unhandled Exception:
System.TypeInitializationException: The type initializer for 'System.Windows.Forms.WindowsFormsSynchronizationContext' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.Windows.Forms.ThemeEngine' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.Drawing.GDIPlus' threw an exception. ---> System.DllNotFoundException: libc assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) System.Drawing.GDIPlus.uname(intptr)
  at System.Drawing.GDIPlus..cctor () [0x0006e] in <48f39c7a28e14dde88053e621a43853e>:0 
   --- End of inner exception stack trace ---
  at System.Drawing.StringFormat..ctor (System.Drawing.StringFormatFlags options, System.Int32 language) [0x00011] in <48f39c7a28e14dde88053e621a43853e>:0 
  at System.Drawing.StringFormat..ctor () [0x00000] in <48f39c7a28e14dde88053e621a43853e>:0 
  at (wrapper remoting-invoke-with-check) System.Drawing.StringFormat..ctor()
  at System.Windows.Forms.ThemeWin32Classic.ResetDefaults () [0x0001f] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.ThemeWin32Classic..ctor () [0x00006] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.ThemeEngine..cctor () [0x00012] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
   --- End of inner exception stack trace ---
  at System.Windows.Forms.SystemInformation.get_MenuAccessKeysUnderlined () [0x00000] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.Control..ctor () [0x000d6] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control..ctor()
  at System.Windows.Forms.WindowsFormsSynchronizationContext..cctor () [0x00000] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
   --- End of inner exception stack trace ---
  at System.Windows.Forms.Control..ctor () [0x0000d] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.ScrollableControl..ctor () [0x00000] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.ContainerControl..ctor () [0x0000e] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.Form..ctor () [0x00012] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.MessageBox+MessageBoxForm..ctor (System.Windows.Forms.IWin32Window owner, System.String text, System.String caption, System.Windows.Forms.MessageBoxButtons buttons, System.Windows.Forms.MessageBoxIcon icon, System.Boolean displayHelpButton) [0x0000c] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.MessageBox+MessageBoxForm..ctor (System.Windows.Forms.IWin32Window owner, System.String text, System.String caption, System.Windows.Forms.MessageBoxButtons buttons, System.Windows.Forms.MessageBoxIcon icon) [0x00000] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.MessageBox+MessageBoxForm..ctor(System.Windows.Forms.IWin32Window,string,string,System.Windows.Forms.MessageBoxButtons,System.Windows.Forms.MessageBoxIcon)
  at System.Windows.Forms.MessageBox.Show (System.String text) [0x00000] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at PSE.CLR_PSE_PluginLog.MsgBoxErrorTrapper (System.Exception e) [0x0003b] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at CLRDEV9.CLR_DEV9.Log_Fatal (System.Exception ex) [0x00000] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at CLRDEV9.CLR_DEV9.Init () [0x00054] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at PSE.CLR_PSE_DEV9.DEV9init () [0x00000] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at (wrapper native-to-managed) PSE.CLR_PSE_DEV9.DEV9init(System.Exception&)`
```

**Comment by TheLastRar** [comment]
That error is a new one, not sure what would cause it

libexif is missing (you should be able to install it via your package manager (is it pacman for Arch?))

Also can you run ldd on libmono-native.so

**Comment by samuk190** [comment]
> That error is a new one, not sure what would cause it
> 
> libexif is missing (you should be able to install it via your package manager (is it pacman for Arch?))
> 
> Also can you run ldd on libmono-native.so

Strange, libexif is installed. (I tried reinstall with pacman -S) still same error on console

output of that ldd
`ldd mono_i386/usr/lib/libmono-native.so 
	linux-gate.so.1 (0xf7ed7000)
	libgssapi_krb5.so.2 => /usr/lib32/libgssapi_krb5.so.2 (0xf7d75000)
	libm.so.6 => /usr/lib32/libm.so.6 (0xf7ca8000)
	librt.so.1 => /usr/lib32/librt.so.1 (0xf7c9d000)
	libpthread.so.0 => /usr/lib32/libpthread.so.0 (0xf7c7b000)
	libc.so.6 => /usr/lib32/libc.so.6 (0xf7a89000)
	libkrb5.so.3 => /usr/lib32/libkrb5.so.3 (0xf79a3000)
	libk5crypto.so.3 => /usr/lib32/libk5crypto.so.3 (0xf7970000)
	libcom_err.so.2 => /usr/lib32/libcom_err.so.2 (0xf796b000)
	libkrb5support.so.0 => /usr/lib32/libkrb5support.so.0 (0xf795a000)
	libdl.so.2 => /usr/lib32/libdl.so.2 (0xf7954000)
	libkeyutils.so.1 => /usr/lib32/libkeyutils.so.1 (0xf794a000)
	libresolv.so.2 => /usr/lib32/libresolv.so.2 (0xf7930000)
	/usr/lib/ld-linux.so.2 (0xf7ed9000)
`

**Comment by samuk190** [comment]
installed libexif (32) 
got new error: 
Init Mono Runtime
Loaded Plugin
The type initializer for 'Sys' threw an exception.
  at System.IO.FileSystem.FileExists (System.ReadOnlySpan`1[T] fullPath, System.Int32 fileType, Interop+ErrorInfo& errorInfo) [0x00007] in <9f0df102fe6e4cfea29d2e46f585d8a5>:0 
  at System.IO.FileSystem.FileExists (System.ReadOnlySpan`1[T] fullPath) [0x00006] in <9f0df102fe6e4cfea29d2e46f585d8a5>:0 
  at System.IO.File.Exists (System.String path) [0x00043] in <9f0df102fe6e4cfea29d2e46f585d8a5>:0 
  at PSE.CLR_PSE_PluginLog.Open (System.String logFolderPath, System.String logFileName, System.String prefix, System.Collections.Generic.Dictionary`2[TKey,TValue] sourceIDs) [0x00078] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at CLRDEV9.CLR_DEV9.LogSetup () [0x00051] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at CLRDEV9.CLR_DEV9.Init () [0x00000] in <fd03a64f865b42448fec49d2c9d711c5>:0 
InnitError

Unhandled Exception:
System.TypeInitializationException: The type initializer for 'System.Windows.Forms.WindowsFormsSynchronizationContext' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.Windows.Forms.ThemeEngine' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.Drawing.GDIPlus' threw an exception. ---> System.DllNotFoundException: libc assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) System.Drawing.GDIPlus.uname(intptr)
  at System.Drawing.GDIPlus..cctor () [0x0006e] in <48f39c7a28e14dde88053e621a43853e>:0 
   --- End of inner exception stack trace ---
  at System.Drawing.StringFormat..ctor (System.Drawing.StringFormatFlags options, System.Int32 language) [0x00011] in <48f39c7a28e14dde88053e621a43853e>:0 
  at System.Drawing.StringFormat..ctor () [0x00000] in <48f39c7a28e14dde88053e621a43853e>:0 
  at (wrapper remoting-invoke-with-check) System.Drawing.StringFormat..ctor()
  at System.Windows.Forms.ThemeWin32Classic.ResetDefaults () [0x0001f] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.ThemeWin32Classic..ctor () [0x00006] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.ThemeEngine..cctor () [0x00012] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
   --- End of inner exception stack trace ---
  at System.Windows.Forms.SystemInformation.get_MenuAccessKeysUnderlined () [0x00000] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.Control..ctor () [0x000d6] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control..ctor()
  at System.Windows.Forms.WindowsFormsSynchronizationContext..cctor () [0x00000] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
   --- End of inner exception stack trace ---
  at System.Windows.Forms.Control..ctor () [0x0000d] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.ScrollableControl..ctor () [0x00000] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.ContainerControl..ctor () [0x0000e] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.Form..ctor () [0x00012] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.MessageBox+MessageBoxForm..ctor (System.Windows.Forms.IWin32Window owner, System.String text, System.String caption, System.Windows.Forms.MessageBoxButtons buttons, System.Windows.Forms.MessageBoxIcon icon, System.Boolean displayHelpButton) [0x0000c] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at System.Windows.Forms.MessageBox+MessageBoxForm..ctor (System.Windows.Forms.IWin32Window owner, System.String text, System.String caption, System.Windows.Forms.MessageBoxButtons buttons, System.Windows.Forms.MessageBoxIcon icon) [0x00000] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.MessageBox+MessageBoxForm..ctor(System.Windows.Forms.IWin32Window,string,string,System.Windows.Forms.MessageBoxButtons,System.Windows.Forms.MessageBoxIcon)
  at System.Windows.Forms.MessageBox.Show (System.String text) [0x00000] in <47c199caaa3c4c6c9c48a7c6ebb3fd29>:0 
  at PSE.CLR_PSE_PluginLog.MsgBoxErrorTrapper (System.Exception e) [0x0003b] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at CLRDEV9.CLR_DEV9.Log_Fatal (System.Exception ex) [0x00000] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at CLRDEV9.CLR_DEV9.Init () [0x00054] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at PSE.CLR_PSE_DEV9.DEV9init () [0x00000] in <fd03a64f865b42448fec49d2c9d711c5>:0 
  at (wrapper native-to-managed) PSE.CLR_PSE_DEV9.DEV9init(System.Exception&)

**Comment by TheLastRar** [comment]
I have, over the weekend, been investigating this in a Arch Linux VM, and found a couple issues;

1.) Flatpak's sandbox is interfering with CLR_DEV9's ability to any dependencies that Flatpak's PCSX2 package doesn't use.

While CLR_DEV9 is able to load it's bundled mono, it still needs to access the system installed version of mono for configuration files in located in ``/etc/mono``, which Flatpak's sandbox prevents.
This can be worked around with a patch to CLR_DEV9 and copying ``/etc/mono`` into ``mono_i386``
Flatpak's sandbox also prevents mono (specificity ``libmono-native.so`` & ``libdgiplus.so```)  from loading any extra dependencies.
This proved somewhat difficult as I couldn't figure out how to get Flatpak to load the extra dependencies alongside PCSX2, In the end I had my plugin force load the needed files (this patch was not pushed to master).

It may have been easier to try and inset the needed files into the runtime package rather than the app package, but that was something I didn't try.

2.) Flatpak's sandbox prevents CLR_DEV9 from accessing the network.

It seems Flatpak requires that the app package require a networking permission before my plugin can access the network. I don't know force pcsx2 to get that permission.

In short, I don't think is reasonably possible to use CLR_DEV9 from within Flatpak's sandbox

**Comment by samuk190** [comment]
> I have, over the weekend, been investigating this in a Arch Linux VM, and found a couple issues;
> 
> 1.) Flatpak's sandbox is interfering with CLR_DEV9's ability to any dependencies that Flatpak's PCSX2 package doesn't use.
> 
> While CLR_DEV9 is able to load it's bundled mono, it still needs to access the system installed version of mono for configuration files in located in `/etc/mono`, which Flatpak's sandbox prevents.
> This can be worked around with a patch to CLR_DEV9 and copying `/etc/mono` into `mono_i386`
> Flatpak's sandbox also prevents mono (specificity `libmono-native.so` & ``libdgiplus.so```) from loading any extra dependencies.
> This proved somewhat difficult as I couldn't figure out how to get Flatpak to load the extra dependencies alongside PCSX2, In the end I had my plugin force load the needed files (this patch was not pushed to master).
> 
> It may have been easier to try and inset the needed files into the runtime package rather than the app package, but that was something I didn't try.
> 
> 2.) Flatpak's sandbox prevents CLR_DEV9 from accessing the network.
> 
> It seems Flatpak requires that the app package require a networking permission before my plugin can access the network. I don't know force pcsx2 to get that permission.
> 
> In short, I don't think is reasonably possible to use CLR_DEV9 from within Flatpak's sandbox
ok.. which version I can use in arch that works? well I guess I have to try downgrade to 32bits version of pcsx2, let me try

**Comment by samuk190** [comment]
@TheLastRar  after some hours of tirelessy trying, I finnally managed it to work!(but with old version x86)
got 1.6.0 package ( pcsx2-1.6.0-2-x86_64.pkg.tar.zst )  and CLR9_DEV worked very nice. Funny that gigahertz plugin had same problem of crashing when opening the game, but gladly your plugin worked with no problems.  Well thanks for the effort 
If someone look for that package just google it and download from archive.org
